apps/api/src/applicant/external-api.service.ts

```diff
@@ -512,7 +514,33 @@ export class ExternalAPIService {
       };
     });
   }
-
+  /**
+   * Jan 2024 - This function fixes an issue caused by the sync in production. There are at least one applicant that have an ats1_id already in use by another applicant.
+   * This is causing an
+   * @param applicants - List of incoming applicants from ATS
+   * @returns A filtered list of applicants that will not cause a violation of the unique key constraint for the ats1_id
+   */
+  async filterContradictoryRows(applicants: AtsApplicant[]) {
+    const existingApplicants: IENApplicant[] = await this.ienapplicantRepository.find({});
+    const contradictions: unknown[] = [];
+    const filteredApplicants = applicants.filter(applicant => {
+      const contradiction = existingApplicants.find(
+        existingApplicant =>
+          existingApplicant.ats1_id === applicant.ats1_id.toString() &&
+          existingApplicant.id !== applicant.applicant_id,
```

Comment:
this is just filtering one of the existing applicants out? aren't they two different applicants or is this referring to same applicant with the same `ats1_id`?

Reply:
This is for finding an existing applicant with the same ats1_id but different applicant id in the system. 
On prod there is at least one row that has this issue, and it's preventing the sync from completing.